runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -42,107 +42,106 @@ uint8_t *J9::ARM64::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterat
+   48,                                       // TR_InlinedVirtualMethodWithNopGuard    = 34
+   48,                                       // TR_InlinedInterfaceMethodWithNopGuard  = 35
+   32,                                       // TR_SpecialRamMethodConst               = 36
+   48,                                       // TR_InlinedHCRMethod                    = 37
+   40,                                       // TR_ValidateStaticField                 = 38
+   40,                                       // TR_ValidateClass                       = 39
+   32,                                       // TR_ClassAddress                        = 40
+   16,                                       // TR_HCR                                 = 41
+   64,                                       // TR_ProfiledMethodGuardRelocation       = 42
+   64,                                       // TR_ProfiledClassGuardRelocation        = 43
    0,                                        // TR_HierarchyGuardRelocation            = 44
    0,                                        // TR_AbstractGuardRelocation             = 45
-   0,                                        // TR_ProfiledInlinedMethodRelocation     = 46
-   0,                                        // TR_MethodPointer                       = 47
-   0,                                        // TR_ClassPointer                        = 48
-   0,                                        // TR_CheckMethodExit                     = 49
-   0,                                        // TR_ValidateArbitraryClass              = 50
-   0,                                        // TR_EmitClass                           = 51
-   0,                                        // TR_JNISpecialTargetAddress             = 52
-   0,                                        // TR_VirtualRamMethodConst               = 53
-   0,                                        // TR_InlinedInterfaceMethod              = 54
-   0,                                        // TR_InlinedVirtualMethod                = 55
+   64,                                       // TR_ProfiledInlinedMethodRelocation     = 46
+   40,                                       // TR_MethodPointer                       = 47
+   32,                                       // TR_ClassPointer                        = 48
+   16,                                       // TR_CheckMethodExit                     = 49
+   24,                                       // TR_ValidateArbitraryClass              = 50
+   0,                                        // TR_EmitClass (not used)                = 51
+   32,                                       // TR_JNISpecialTargetAddress             = 52
+   32,                                       // TR_VirtualRamMethodConst               = 53
+   48,                                       // TR_InlinedInterfaceMethod              = 54
+   48,                                       // TR_InlinedVirtualMethod                = 55
```

Comment:
These should be 40; on all the other 64-bit platforms it is 40. Additionally, in what will eventually be the one and only table of sizes (work in progress), the size of the binary template for also ends up evaluating to 40 (if you follow the inheritance chain):
https://github.com/eclipse/openj9/blob/0739a278d3a5cbc0c19e14e7fcb4d8164379b9ea/runtime/compiler/runtime/RelocationRecord.cpp#L4668-L4669

Reply:
It seems I overlooked the `if` below, and over-counted the sizes for those two types.
https://github.com/eclipse/openj9/blob/d01bf6fab7100415c9c78341cfefd95848610888/runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp#L417-L418